src/basic/utf8.c

```diff
@@ -1,26 +1,10 @@
 /* SPDX-License-Identifier: LGPL-2.1-or-later */
 
-/* Parts of this file are based on the GLIB utf8 validation functions. The
- * original license text follows. */
-
-/* gutf8.c - Operations on UTF-8 strings.
+/* Parts of this file are based on the GLIB utf8 validation functions. The original copyright follows.
  *
+ * gutf8.c - Operations on UTF-8 strings.
  * Copyright (C) 1999 Tom Tromey
  * Copyright (C) 2000 Red Hat, Inc.
- *
- * This library is free software; you can redistribute it and/or
- * modify it under the terms of the GNU Library General Public
```

Comment:
Uuuuh this file is lgpl but this mentions gpl... this is not good, we need to audit it, can you please leave it for now

Reply:
However given it's a different license than the file, although compatible, we should keep a reference

Reply:
Hmm, I think we should go in the other direction. Still drop the boilerplate text, but change the  SPDX-License-Identifier to LGPL-2.0-or-later.

Reply:
@keszybz You mean GPL, rather than LGPL?

Reply:
No this file is LGPL-2.0-or-later given there's a code snippet  under that license

Reply:
Done. Also the README file is updated.